Power laser port 2 electric bogaloo

Migratingcocofruit opened the pull request:
<!-- By ticking or leaving ticked the option "Allow edits and access to secrets by maintainers" you give permission for repository maintainers to push changes to your branch without explicitly asking. -->

<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## What Does This PR Do
Ports the power transmission laser again, this time with 50% more description and 100% less broken behavior(let me dream okay).

The Power Transmission Laser sells the excess power produced by the station for credits.
Credits earned per cycle approach 0 as the output power approaches 0, and approaches 10 as the power output approaches infinity.
75% of the money generated goes to engineering, and the remaining 25% go to cargo. 
The PTL will send money when it reaches 200 credits of stored funds so it doesn't spam the accounts with tiny transactions

The PTL will send updates as the station sends more and more energy to central, the threshold for a new announcement rises exponentially capped at 200GJ above the last one.

The PTL produces a beam that is purely visual now

The bore of the PTL will produce flashes from 5 MW of output power and up. The flash strength is output power in megawatts divided by 5 and capped by 3. Looking directly at the bore(opposite the direction of the PTL) gives the full flash effect, looking away from it gives no flash, and looking at it partly multiplies the flash strength by the cosine of the difference between the direction you're looking at and a line from you to the bore. The range scales with power in the same way and caps at 8 tiles away, and you can't get flashed while standing behind the bore(indicated by a black and yellow line in the PTL room)

Radiation starts at 30MW with the pulse strength being 50 times output power in megawatts divided by 30. This isn't capped, but should produce only a moderate amount of radiation even when outputting a gigawatt.

The PTL can be used to target megafauna in the mining base and it's immediate vicinity, dealing 10 * output in megawatts fire damage. 
Megafauna damaged by the laser lose their loot, and megafauna that are killed by the laser also get dusted.
Once the targeted megafauna is dead the laser will automatically revert to selling power to central.

At the end of the round the total amount of energy exported by the PTL(in GigaJoules) is presented

PTL locations:

Cyberiad
![image](https://github.com/user-attachments/assets/c8e40923-807e-4b08-b50d-d47b4f481271)

Kerberos
![image](https://github.com/user-attachments/assets/0cdbccbb-d96f-49a8-9833-3a53fbfe32ea)

Cerebron
![image](https://github.com/user-attachments/assets/f44b43a9-4ea0-4f4b-aa2d-8f7f52836ace)

Farragus
![image](https://github.com/user-attachments/assets/a917660c-e1e5-479c-b18c-12f788b16c4d)

Diagoras
![image](https://github.com/user-attachments/assets/1198f45d-dd2b-4e73-9dd4-056d6f25683d)

<!-- Include a small to medium description of what your PR changes. -->
<!-- Document all changes, as not doing this may delay reviews or even discourage maintainers from merging your PR! -->
<!-- If your PR fixes an issue, add "Fixes #1234" somewhere in the PR description. This will automatically close the bug upon PR submission. -->

## Why It's Good For The Game
Allows engineers to do something with all the power they make.
<!-- Add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Images of changes
![380451347-0a939632-d3f6-4bbb-8d63-fb12a1022d36](https://github.com/user-attachments/assets/cc86a694-a486-4323-b42d-6daa35ad544e)
![image](https://github.com/user-attachments/assets/d56aba4b-b972-439b-85f0-cc1313afad4b)

<!-- If you did not make a map or sprite edit, you may delete this section. You may include a gif or mp4 of your feature if you want. -->

## Testing
So far:
- Observed the correct amount of payout being given at the correct threshold.
- Ordered the PTL crate and used it to build one
- Rotated the PTL and made sure it rotates when it should, and not when it shouldn't
- Spawned lavaland mobs in the mining base and blasted them. They did not drop loot when they died.
- Got flashed and irradiated by the PTL
<!-- How did you test the PR, if at all? -->

<hr>

## ToDo
- [x] Make the blocker ref into a stored UID value.
- [x] Make fine adjustment better for low power use
- [x] Have max capacity not be infinite
- [x] Make the laser deconstructable and give it components and a cargo crate.
- [x] Add missing stuff to the rooms
- [x] Make the rooms not ugly
- [x] Get the PTL to show up properly on the map editor
- [x] Implement bombardment of megafauna close to the mining base
- [x] Delete megafauna loot if they are killed by the laser

### Declaration
![image](https://github.com/user-attachments/assets/4244bc18-066f-433b-a9f9-3b506a3eb782)

- [x] I confirm that I either do not require [pre-approval](https://github.com/ParadiseSS13/Paradise/blob/master/docs/CODE_OF_CONDUCT.md#types-of-changes-that-need-approval) for this PR, or I have obtained such approval and have included a screenshot to demonstrate this below.
  <!-- Replace the box with [x] to mark as complete. -->
  <!-- Ensure there are no spaces between the x and the square brackets [] else this will not work properly. -->
  <hr>

## Changelog

:cl:
add: Ports the power transmission laser, a machine that sells the excess power produced by the station for credits
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->
<!-- If a PR has no impact on players (i.e. a code refactor that does not change functionality) then the entire Changelog heading and contents can be removed. -->

Henri215 commented:
Implement the fixes from https://github.com/ParadiseSS13/Paradise/pull/27167 and https://github.com/ParadiseSS13/Paradise/pull/27169

Migratingcocofruit commented:
> Implement the fixes from #27167 and #27169

Already did

Henri215 commented:
> > Implement the fixes from #27167 and #27169
> 
> Already did

And you did commit it? Because i am seeing the code from before the fixes.

Henri215 commented on code/goonstation/modules/power/power_transmission_laser/transmission_laser.dm:
```diff
@@ -0,0 +1,508 @@
+
+/obj/machinery/power/transmission_laser/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>Laser currently has [unsent_earnings] unsent credits.\s<span/>"
+	. += "<span class='notice'>Laser has generated [total_earnings] credits.\s<span/>"
+	. += "<span class='notice'>Laser has sold [total_energy] Joules\s<span/>"
+
```
```suggestion
	. += "<span class='notice'>Laser currently has [unsent_earnings] unsent credits.<span/>"
	. += "<span class='notice'>Laser has generated [total_earnings] credits.<span/>"
	. += "<span class='notice'>Laser has sold [total_energy] Joules.<span/>"
```
Theres this one from the fix too.

Migratingcocofruit replied:
done

Henri215 reviewed (commented):
The others will be sending more stuff to be fixed soon.

S34NW commented:
Here was some feedback I was given by dev team members, apologies if anything duplicates what you've done:

- You're using both pixel_y/pixel_x and the multitile component. You can make it so you only need the multitile component
- The `var/static/announcement_threshold` can probably be  changed to a normal variable
- `var/atom/blocker` should really be a UID instead of a ref.
- get_back_turf() isn't necessary with a good multitile component
- This should be rebuildable/deconstructable in case of repairs
- Ghosts cant view the TGUI
- beams through cerestation disposals leading to items either being heavily damaged or you know gibbing disposaled people
- The fact it is a beam that makes space travel around it impossible (without changing zlvls) rather than just beaming the power through the roof.
- The map roomed are silly.
- Machine is indestructible.
- Mapped in roundstart (preference)
- Doesn't look proper in the map editor (also linked to the issue that it doesnt properly use multitile)
- Its already mapped in, I believe most people thought it would be something you had to build
- Money is kinda underwhelming for how much you have to put into it
- Gib + explosion laser is too powerful
-  the knob hardly moves <10mw
![delwedd](https://github.com/user-attachments/assets/6b449ad5-1bdb-40ca-ad58-1248f6d194ab)
- you can set the 'output' power to whatever you want, and insta gib/destroy any items that're in the path, even if it lacks the power to do so
- You can do funny things like this (it doesn't generate any more cash, and will automatically shut down but the beam will vaporise you)
![delwedd](https://github.com/user-attachments/assets/7a566330-65ec-40d7-bfaf-cf9926d62394)
- the new rooms are giant bare boxes with no decorations or contents. if we're adding an entirely new engineering area, it needs to be thematic, it needs to look like it belongs in engineering. what items should be in it? any lockers? storage? warning signs on the walls, warning decals around the laser?
- it almost certainly doesn't need reinforced walls.
- on cere it needs to be put somewhere it won't affect the cere waste loop, even if that means altering the waste loop path or putting the laser further out and adding disposals pipes where the new area intersects the waste loop
- the room needs to be more structurally integrated into the rest of the station. the overall shape of the perimeter should be followed more closely. i know box gravgen is another big boxy room but that sucks too and we don't want to keep making it worse
- fire alarms are missing on box and cere
- there's no cameras

1080pCat commented:
> * The fact it is a beam that makes space travel around it impossible (without changing zlvls) rather than just beaming the power through the roof.

i quite like this, it makes space more risky, and requires communication. however the PTL mirrors should be added/ported so the beam can be snaked around, so it doesn't need to be in a odd spot.

1080pCat commented:
Another thing. I personally think the how the BSH handled the power level/input before you reworked it would be the best thing for handling the power
![image](https://github.com/user-attachments/assets/ca69dba8-e8be-43fa-a209-22057dc3df58)

Having a slider bar go from 0-1000 (kw, mw, gw) with the green bar showing the grid saturation/reserve energy would be nice.
(more of a wiki thing than anything) but the reserve power makes /no sense/ to me, i could never get the bar to move, the GJ would go up very slowly, but the bar would never move. I dunno if this was a bug, or just misunderstood mechanics.

Migratingcocofruit commented:
> Here was some feedback I was given by dev team members, apologies if anything duplicates what you've done:
> 
>     * You're using both pixel_y/pixel_x and the multitile component. You can make it so you only need the multitile component

pixel_x and pixel_y move the sprite while the multitile component adds invisible dense structures that only impede movement. The BSH and DNA vault also do this. 
The reason it doesn't look right in the map editor is because the machine itself is actually only one tile, and that tile needs a wire node under it. 
The offsets are used so that the machine could be rotated around that tile, preventing it from getting disconnected from the powernet if it is rotated for any reason.

>     * The `var/static/announcement_threshold` can probably be  changed to a normal variable
I was asked to make it a static in the previous PR. And considering the announcement are about the power sent to centcom, rather than the power output by the PTL it makes sense that they would be shared between them.

>     * `var/atom/blocker` should really be a UID instead of a ref.
Yeah, I think I can make it work.
> 
>     * get_back_turf() isn't necessary with a good multitile component
> 
Get back turf isn't called at all. I forgot to delete it.

>     * This should be rebuildable/deconstructable in case of repairs
> 
I'll make it so. And make a cargo crate for it as well.

>     * Ghosts cant view the TGUI
> 
Fixed that already
>     * beams through cerestation disposals leading to items either being heavily damaged or you know gibbing disposaled people
> 
Oh yeah. I'll take care of that.
>     * The fact it is a beam that makes space travel around it impossible (without changing zlvls) rather than just beaming the power through the roof.
> 
I'll get the mirrors as well, or maybe just have it done with reflectors.

>     * The map roomed are silly.
> 
:(
>     * Machine is indestructible.
>
duplicate.  
>     * Mapped in roundstart (preference)
> 
I'd rather have it up at roundstart. This is a good default project for engineering and I'd rather it requires as little effort to setup as possible, with most of the effort being put into power production. Minimum effort credits will likely need adjustment, but I think playtesting would be the best way to figure it out.
>     * Doesn't look proper in the map editor (also linked to the issue that it doesnt properly use multitile)
> 
It does use multitile properly. Again it's due to having it rotate around the wire node under it.

>     * Its already mapped in, I believe most people thought it would be something you had to build
> 
Again, duplicate.
>     * Money is kinda underwhelming for how much you have to put into it
> 
You can get around 7 credits per second with a powerfull SM setup. This doesn't sound like much, but it comes up to 25,200 credits per hour.
>     * Gib + explosion laser is too powerful
> 
I don't really know how powerfull the laser effects should be in the end and at what thresholds. Open to suggestions on that
>     * the knob hardly moves <10mw

Are logarithmic knobs a thing? I'll try to figure something out.

>     * you can set the 'output' power to whatever you want, and insta gib/destroy any items that're in the path, even if it lacks the power to do so
> 
>     * You can do funny things like this (it doesn't generate any more cash, and will automatically shut down but the beam will vaporise you)

Fixed that 

>     * the new rooms are giant bare boxes with no decorations or contents. if we're adding an entirely new engineering area, it needs to be thematic, it needs to look like it belongs in engineering. what items should be in it? any lockers? storage? warning signs on the walls, warning decals around the laser?
> 
>     * it almost certainly doesn't need reinforced walls.
> 
>     * on cere it needs to be put somewhere it won't affect the cere waste loop, even if that means altering the waste loop path or putting the laser further out and adding disposals pipes where the new area intersects the waste loop
> 
>     * the room needs to be more structurally integrated into the rest of the station. the overall shape of the perimeter should be followed more closely. i know box gravgen is another big boxy room but that sucks too and we don't want to keep making it worse
> 
>     * fire alarms are missing on box and cere
> 
>     * there's no cameras

I'll fix the obvious deficiencies now, but I may need some guidance in regards to decoration and overall aesthetic design.

I'll make a to-do list in the PR comment for all the stuff.

Migratingcocofruit commented:
> Another thing. I personally think the how the BSH handled the power level/input before you reworked it would be the best thing for handling the power ![image](https://private-user-images.githubusercontent.com/96908085/379270367-ca69dba8-e8be-43fa-a209-22057dc3df58.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk2ODc3MDIsIm5iZiI6MTcyOTY4NzQwMiwicGF0aCI6Ii85NjkwODA4NS8zNzkyNzAzNjctY2E2OWRiYTgtZThiZS00M2ZhLWEyMDktMjIwNTdkYzNkZjU4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDIzVDEyNDMyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTczMGM0NWQ5YzdiMjk5ZjJiYWY1ODk1ZDZiMDZkZWEwOTM0Y2FkNWU5NTM3MzIxZjRiOGI0MTVjMDI1NWUyMTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Zt1gfgsAzDHsQICW-_H-9RhZ1Q5Te6cIXTbyDzv419Y)
> 
> Having a slider bar go from 0-1000 (kw, mw, gw) with the green bar showing the grid saturation/reserve energy would be nice. (more of a wiki thing than anything) but the reserve power makes /no sense/ to me, i could never get the bar to move, the GJ would go up very slowly, but the bar would never move. I dunno if this was a bug, or just misunderstood mechanics.

That's because the maximum value is infinity. I need to figure out where to put the limit on that as well, I suppose.

Henri215 commented:
"add: Ports the power transmission laser, a machinery that sells the excess power produced by the station for credits"

This changelog would give the players more information.

Migratingcocofruit commented:
Reflection is deemed problematic, so I won't implement it.

Henri215 commented on code/goonstation/modules/power/power_transmission_laser/transmission_laser.dm:
```diff
@@ -0,0 +1,468 @@
+
+/obj/machinery/power/transmission_laser/screwdriver_act(mob/living/user, obj/item/I)
+	if(firing)
+		to_chat(user,"<span class='info'>Turn the laser off first<span/>")
```
```suggestion
		to_chat(user,"<span class='info'>Turn the laser off first.<span/>")
```

Henri215 commented on code/goonstation/modules/power/power_transmission_laser/transmission_laser.dm:
```diff
@@ -0,0 +1,468 @@
+		qdel(listed_beam)
+
+// Beam
+/obj/effect/transmission_beam
```
It cant be a /obj/effect/ebeam for what you want to do with it?

Migratingcocofruit replied:
No. Following a discussion with balance and design the beam is going to be a purely visual effect that goes under everything. Making it an ebeam will only make it more complicated.

warriorstar-orion reviewed (changes requested):
Map changes looking good.

## Cere

Move these airlocks down so that the waste loop area is walled off completely. Also move/remove this floating lightbulb:

![image](https://github.com/user-attachments/assets/db1af844-ba02-4a14-a8a0-6cd5b3a1b04a)

Extra camera, use plating under window spawners, light tube embedded in wall, use flooring under airlocks, floating light bulb in front of first set of airlocks, move grille away from in front of first set of airlocks:

![image](https://github.com/user-attachments/assets/98a7301e-c66d-40e6-912b-24d50ee49a42)

Pixel-shift these signs onto the wall from the tile in front of them, since they can't be seen from the other side:

![image](https://github.com/user-attachments/assets/f0f44c04-7e86-432f-b8b4-334bd79dcf4f)

Move the gap in the waste belt to directly in front of the PTL beam path:

![image](https://github.com/user-attachments/assets/1c640980-2fff-48ec-b58b-1b44653c258c)

The perimeter of any rock walls that can be dug directly to space need to be of type `/turf/simulated/mineral/ancient/outer`:

![image](https://github.com/user-attachments/assets/a01719af-267d-435e-a5a2-1ddb1f017ac0)

## Delta

The walling here is a little awkward; you want to avoid double-walling (in pink), and this gap in space feels out of place (in blue):

![image](https://github.com/user-attachments/assets/51ca5f71-f586-4fb4-8371-cebed1e65af4)

I would pull both of these to the adjacent walls, and use the extra space to decorate a bit more, something along these lines:

![2024_10_26__11_27_37__paradise dme  deltastation dmm  - StrongDMM](https://github.com/user-attachments/assets/f53c7dbc-4d12-41c6-983e-dabdf160761e)

Duplicate cameras on this station too.

## Meta

Use plating under window spawners, and push these vents/scrubbers so that they're not directly under lights. Or move the lights.

![image](https://github.com/user-attachments/assets/84c3c35d-1774-41a1-a719-d376d267a4b4)

## Emerald

Floating light tube in front of airlock and in front of rack, use plating under window spawners, and move vents/scrubbers away from directly under lights.

![image](https://github.com/user-attachments/assets/4263bbb1-02c7-4348-9761-e4bb5ccc8f7f)

These walls and windows bumping up against each other need to be adjusted. Either pull the rwall towards the airlock and remove the line of windows, or vice-versa.

![image](https://github.com/user-attachments/assets/b213f1ed-f941-4539-9447-f95cf515f2f8)

Either remove this camera (these rooms don't need two cameras), or remove the other one and move this off the wall that has the APC on it.

## General

These areas would still benefit from light decoration in the form of colored tiles and decals. This is an idea how you may want to decorate things, but get creative:

![2024_10_26__10_53_39__paradise dme  boxstation dmm  - StrongDMM](https://github.com/user-attachments/assets/c0aa493e-6c2a-4b6f-ab6e-b187834309ba)

Especially note the station-specific theming. On Box, engineering doesn't use colored tiles extensively, but they are used directly surrounding airlocks.

Migratingcocofruit commented:
> Map changes looking good.
> ## Cere
> 
> Move these airlocks down so that the waste loop area is walled off completely. Also move/remove this floating lightbulb:
> 
> ![image](https://private-user-images.githubusercontent.com/59303604/380431414-db1af844-ba02-4a14-a8a0-6cd5b3a1b04a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk5NzAwMDgsIm5iZiI6MTcyOTk2OTcwOCwicGF0aCI6Ii81OTMwMzYwNC8zODA0MzE0MTQtZGIxYWY4NDQtYmEwMi00YTE0LWE4YTAtNmNkNWIzYTFiMDRhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI2VDE5MDgyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM2ZjgwODAzMTJlMTkzODNlMTExNTIzNTA4NDIxN2Q0M2I5NzMyM2JiNTAwN2FhYmRkZmZiOTEwNTRmZDIyMmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.5dTBpPOpYdfSSeB1rb5f-ROvOLYwFOa9nMQ9X-_NoiQ)
> 
> Extra camera, use plating under window spawners, light tube embedded in wall, use flooring under airlocks, floating light bulb in front of first set of airlocks, move grille away from in front of first set of airlocks:
> 
> ![image](https://private-user-images.githubusercontent.com/59303604/380432196-98a7301e-c66d-40e6-912b-24d50ee49a42.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk5NzAwMDgsIm5iZiI6MTcyOTk2OTcwOCwicGF0aCI6Ii81OTMwMzYwNC8zODA0MzIxOTYtOThhNzMwMWUtYzY2ZC00MGU2LTkxMmItMjRkNTBlZTQ5YTQyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI2VDE5MDgyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE4OTA4MTk2MjAxZDhmNTBkNjFkMDY2NjcyZjNhNDgxZjc4ZTkxMzU0NDRlZTU2MzAyZDgxMTA3ZDI0OTQ4MTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.jFFCApeTBROacaoTYnvQ4RHIvk-HnpvpkzBfOLQv3ws)
> 
> Pixel-shift these signs onto the wall from the tile in front of them, since they can't be seen from the other side:
> 
> ![image](https://private-user-images.githubusercontent.com/59303604/380432524-f0f44c04-7e86-432f-b8b4-334bd79dcf4f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk5NzAwMDgsIm5iZiI6MTcyOTk2OTcwOCwicGF0aCI6Ii81OTMwMzYwNC8zODA0MzI1MjQtZjBmNDRjMDQtN2U4Ni00MzJmLWI4YjQtMzM0YmQ3OWRjZjRmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI2VDE5MDgyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVhNjU4NGI2ZjI3YTMyMTk1OTBiM2NmOGUwM2QwZTJmMzE0ZDE2NmQ1OTRkNTY2MDc1MTIyZTg4YjAzYTVkYWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.wFe4xOIcOvuE2M6e3I3ncDTk7n2WOdCgDeLcsDEGSpM)
> 
> Move the gap in the waste belt to directly in front of the PTL beam path:
> 
> ![image](https://private-user-images.githubusercontent.com/59303604/380432657-1c640980-2fff-48ec-b58b-1b44653c258c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk5NzAwMDgsIm5iZiI6MTcyOTk2OTcwOCwicGF0aCI6Ii81OTMwMzYwNC8zODA0MzI2NTctMWM2NDA5ODAtMmZmZi00OGVjLWI1OGItMWI0NDY1M2MyNThjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI2VDE5MDgyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQwZjg3MTFjODFiN2UxNmU5OGJmYmRjNmZkMmM4MmEyZTRkMTE3MGI0NTI0YjUxYTYyNGM2MjU0NTE3NjExMWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.e3jJ_EE_fMboNyJVtExCnv36lBtQ7oFoS9nDekMwKJE)
> 
> The perimeter of any rock walls that can be dug directly to space need to be of type `/turf/simulated/mineral/ancient/outer`:
> 
> ![image](https://private-user-images.githubusercontent.com/59303604/380432783-a01719af-267d-435e-a5a2-1ddb1f017ac0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk5NzAwMDgsIm5iZiI6MTcyOTk2OTcwOCwicGF0aCI6Ii81OTMwMzYwNC8zODA0MzI3ODMtYTAxNzE5YWYtMjY3ZC00MzVlLWE1YTItMWRkYjFmMDE3YWMwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI2VDE5MDgyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVkOGU4MDAzYjI0M2M3MDA0NTIwNGY0NGMzZTg0NGJlMjc2OTllODhhYTMzODI3NzhhM2MyYjQwMDU0MmIwMDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.2Xkd2UFx4pkv7YlA0a6Ig3aArlbU0MgjMVfVc3xv_Gs)
> ## Delta
> 
> The walling here is a little awkward; you want to avoid double-walling (in pink), and this gap in space feels out of place (in blue):
> 
> ![image](https://private-user-images.githubusercontent.com/59303604/380433550-51ca5f71-f586-4fb4-8371-cebed1e65af4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk5NzAwMDgsIm5iZiI6MTcyOTk2OTcwOCwicGF0aCI6Ii81OTMwMzYwNC8zODA0MzM1NTAtNTFjYTVmNzEtZjU4Ni00ZmI0LTgzNzEtY2ViZWQxZTY1YWY0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI2VDE5MDgyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFjMzc5MzkwYjk4YzZmM2JlODNmNjkzZDRlZDY2OTViZjE5YTkzNmFiZTk5NmExNjQ1ZThmYTczNjI2ZjZlYjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.D7Z6U9Lvv9-FGCAOy5wodyTdaEPITXSfMg-JYV5Zu0E)
> 
> I would pull both of these to the adjacent walls, and use the extra space to decorate a bit more, something along these lines:
> 
> ![2024_10_26__11_27_37__paradise dme deltastation dmm - StrongDMM](https://private-user-images.githubusercontent.com/59303604/380433578-f53c7dbc-4d12-41c6-983e-dabdf160761e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk5NzAwMDgsIm5iZiI6MTcyOTk2OTcwOCwicGF0aCI6Ii81OTMwMzYwNC8zODA0MzM1NzgtZjUzYzdkYmMtNGQxMi00MWM2LTk4M2UtZGFiZGYxNjA3NjFlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI2VDE5MDgyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY2ZTQ1MDc5N2Y0ZDA1OTFhMDI1MWQzZjAyYWZmZDA0ODA3NDZiZjVhOTgwNGM2MjA4NWI2MmUyOTBlMjFjNjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0._iwqV0Wv0n2QbIs03VlBDpWz0RDeUuu7K15CwkS5Res)
> 
> Duplicate cameras on this station too.
> ## Meta
> 
> Use plating under window spawners, and push these vents/scrubbers so that they're not directly under lights. Or move the lights.
> 
> ![image](https://private-user-images.githubusercontent.com/59303604/380433836-84c3c35d-1774-41a1-a719-d376d267a4b4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk5NzAwMDgsIm5iZiI6MTcyOTk2OTcwOCwicGF0aCI6Ii81OTMwMzYwNC8zODA0MzM4MzYtODRjM2MzNWQtMTc3NC00MWExLWE3MTktZDM3NmQyNjdhNGI0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI2VDE5MDgyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM2NmQ0NDc3MzEwMGE0ZjM5Y2E3OGNjZmMxZTNiNGJhMDVhOGM2ZmYzM2NlZTMyOTYyMmQ5NzU4NWY5MDRlMDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.WJvJQ7cc71t3ytzd2yVSYEfjGw74GJGsrgGpxU3tEfU)
> ## Emerald
> 
> Floating light tube in front of airlock and in front of rack, use plating under window spawners, and move vents/scrubbers away from directly under lights.
> 
> ![image](https://private-user-images.githubusercontent.com/59303604/380433972-4263bbb1-02c7-4348-9761-e4bb5ccc8f7f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk5NzAwMDgsIm5iZiI6MTcyOTk2OTcwOCwicGF0aCI6Ii81OTMwMzYwNC8zODA0MzM5NzItNDI2M2JiYjEtMDJjNy00MzQ4LTk3NjEtZTRiYjVjY2M4ZjdmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI2VDE5MDgyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQwZDA1NmY5MDQwOTEwNzhmNTk5MjdiMGVjYjhmMzVjZTRiOWYwMzMzOTk1YzNlNWJjYmNlNGYxMTJiM2RkY2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.cC1p86zM6abhwy1tyD2MJ87ak1CUNnhkp3xV-UYFcog)
> 
> These walls and windows bumping up against each other need to be adjusted. Either pull the rwall towards the airlock and remove the line of windows, or vice-versa.
> 
> ![image](https://private-user-images.githubusercontent.com/59303604/380434087-b213f1ed-f941-4539-9447-f95cf515f2f8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk5NzAwMDgsIm5iZiI6MTcyOTk2OTcwOCwicGF0aCI6Ii81OTMwMzYwNC8zODA0MzQwODctYjIxM2YxZWQtZjk0MS00NTM5LTk0NDctZjk1Y2Y1MTVmMmY4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI2VDE5MDgyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFkNjFmZDFmZTQ3ZWU2ZTllNTcxYjMxOWM1YjcwMTlmYWFmYzNjNWViNGQxZWZkNjVlZGIyZDkwNDU4ODAxYjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.HUiZC45iGumJRddxk_35jhnR-hUNYN6z46ZixJO7feg)
> 
> Either remove this camera (these rooms don't need two cameras), or remove the other one and move this off the wall that has the APC on it.
> ## General
> 
> These areas would still benefit from light decoration in the form of colored tiles and decals. This is an idea how you may want to decorate things, but get creative:
> 
> ![2024_10_26__10_53_39__paradise dme boxstation dmm - StrongDMM](https://private-user-images.githubusercontent.com/59303604/380433004-c0aa493e-6c2a-4b6f-ab6e-b187834309ba.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk5NzAwMDgsIm5iZiI6MTcyOTk2OTcwOCwicGF0aCI6Ii81OTMwMzYwNC8zODA0MzMwMDQtYzBhYTQ5M2UtNmMyYS00YjZmLWFiNmUtYjE4NzgzNDMwOWJhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI2VDE5MDgyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVhMTFlYjIxOGUyNWY5ODEwN2ZjM2Q5NGJkZjllN2E5MDIyNmQ2ZGM0ZDFlZDZiOGZlOGFhNzk2N2IwZmM2YzMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.rAHQ6SYZKr49c5MooMVMtcX0UzMyCbOphrtUrNEkNEM)
> 
> Especially note the station-specific theming. On Box, engineering doesn't use colored tiles extensively, but they are used directly surrounding airlocks.

Faragus changes pushed

Migratingcocofruit commented:
> Map changes looking good.

> ## Delta
> 
> The walling here is a little awkward; you want to avoid double-walling (in pink), and this gap in space feels out of place (in blue):
> 
> ![image](https://private-user-images.githubusercontent.com/59303604/380433550-51ca5f71-f586-4fb4-8371-cebed1e65af4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk5NzI4MzYsIm5iZiI6MTcyOTk3MjUzNiwicGF0aCI6Ii81OTMwMzYwNC8zODA0MzM1NTAtNTFjYTVmNzEtZjU4Ni00ZmI0LTgzNzEtY2ViZWQxZTY1YWY0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI2VDE5NTUzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWEzOTc5MDhjMmUwNDdkNTVlNDJkNjY1NzE3YzA0Y2M1Y2IzNjBlZjRhNTY3M2YzZWFjMjNkOWI3MGQ3ZDkxMTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.YeHLtKolNKHX1Sy1Xy-WVk-J_rkUOkKtGIMPr5arQ7k)
> 
> I would pull both of these to the adjacent walls, and use the extra space to decorate a bit more, something along these lines:
> 
> ![2024_10_26__11_27_37__paradise dme deltastation dmm - StrongDMM](https://private-user-images.githubusercontent.com/59303604/380433578-f53c7dbc-4d12-41c6-983e-dabdf160761e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk5NzI4MzYsIm5iZiI6MTcyOTk3MjUzNiwicGF0aCI6Ii81OTMwMzYwNC8zODA0MzM1NzgtZjUzYzdkYmMtNGQxMi00MWM2LTk4M2UtZGFiZGYxNjA3NjFlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI2VDE5NTUzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJjMzU0YmQ4ZGNjOTQ1MDAyNTQwNjg0ZWVlYTEzYmVkMDExNWE1YzdjOWEyNjI3MWVmNmRkNDc5YTRlMjUzYTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.uw_8ph_KIsYehw31wblDD63aPcxD6dmrvz_rBz06NkM)
> 
> Duplicate cameras on this station too.

Delta Changes pushed

Migratingcocofruit commented:
Diagoras changes pushed

Migratingcocofruit commented:
Meta changes pushed

Migratingcocofruit commented:
Cyberiad changes pushed

warriorstar-orion reviewed (changes requested):
Map changes look great. I like the visual distinctiveness of each station's room in terms of decorations.

On Cere:

Scrubber pipe extends out here:

![2024_10_28__01_04_32__paradise dme  cerestation dmm  - StrongDMM](https://github.com/user-attachments/assets/47385eea-d180-4687-90be-f75f71af036f)

Catwalk hidden under this rock wall:

![2024_10_28__01_02_39__paradise dme  cerestation dmm  - StrongDMM](https://github.com/user-attachments/assets/ff6ffc33-4966-467a-a7f1-85597dbac05d)

All of these areas need to be updated to align back up with the substation and halls:

![2024_10_28__01_02_22__paradise dme  cerestation dmm  - StrongDMM](https://github.com/user-attachments/assets/7f3aefc2-bdb6-4627-9dac-67b420d8d11e)

Stray unexplored mining area in the hall:

![2024_10_28__01_02_10__paradise dme  cerestation dmm  - StrongDMM](https://github.com/user-attachments/assets/51af909d-13d7-4bc5-96d2-8dfb8f6f6a7a)

Migratingcocofruit commented:
> Map changes look great. I like the visual distinctiveness of each station's room in terms of decorations.
> 
> On Cere:
> 
> Scrubber pipe extends out here:
> 
> All of these areas need to be updated to align back up with the substation and halls:
> 
> Stray unexplored mining area in the hall:

Done

Onyxarias commented:
Currently placing a soft objection on this. The PR looks great from a Design point of view, but I would rather it not have so many broadcasted messages. Every other station goal doesn't do this except for sure emergencies. Instead, I'd like to see the messages restricted down to, say, just megafauna kills. Great work, by the way!

Migratingcocofruit commented:
> Currently placing a soft objection on this. The PR looks great from a Design point of view, but I would rather it not have so many broadcasted messages. Every other station goal doesn't do this except for sure emergencies. Instead, I'd like to see the messages restricted down to, say, just megafauna kills. Great work, by the way!

I get what you mean, but I still want engineering to be able to show off. This is part of the point. Would changing the nature of the announcement be enough?

Onyxarias commented:
> > Currently placing a soft objection on this. The PR looks great from a Design point of view, but I would rather it not have so many broadcasted messages. Every other station goal doesn't do this except for sure emergencies. Instead, I'd like to see the messages restricted down to, say, just megafauna kills. Great work, by the way!
> 
> I get what you mean, but I still want engineering to be able to show off. This is part of the point. Would changing the nature of the announcement be enough?

While true, myself and others have found the constant messages rather annoying, and redundant. If you compare it to another engineering machine, the Supermatter, even that only outputs messages during critical emergencies or anomalies, and the latter is only on the engineering channel. As well, Cargo doesn't have computers loudhailing when they sell certain amounts of plasma.

If engineering REALLY wants to show off, the CE can use the station announcements to relay their accomplishments to the station; it's not something that needs to be done automatically and _constantly._

Again, PR is great, and even with my objection will probably become another fantastic feature for a department that sorely needs it. But I don't think spamming the station's radio is a thing it should do.

Migratingcocofruit commented:
> > > Currently placing a soft objection on this. The PR looks great from a Design point of view, but I would rather it not have so many broadcasted messages. Every other station goal doesn't do this except for sure emergencies. Instead, I'd like to see the messages restricted down to, say, just megafauna kills. Great work, by the way!
> > 
> > 
> > I get what you mean, but I still want engineering to be able to show off. This is part of the point. Would changing the nature of the announcement be enough?
> 
> While true, myself and others have found the constant messages rather annoying, and redundant. If you compare it to another engineering machine, the Supermatter, even that only outputs messages during critical emergencies or anomalies, and the latter is only on the engineering channel. As well, Cargo doesn't have computers loudhailing when they sell certain amounts of plasma.
> 
> If engineering REALLY wants to show off, the CE can use the station announcements to relay their accomplishments to the station; it's not something that needs to be done automatically and _constantly._
> 
> Again, PR is great, and even with my objection will probably become another fantastic feature for a department that sorely needs it. But I don't think spamming the station's radio is a thing it should do.

If everything else was also doing it the radio would be unusable. Good point, that has me convinced. I'll be axing the announcements, and give some more info on the UI.

Burzah commented on code/goonstation/modules/power/power_transmission_laser/transmission_laser.dm:
```diff
@@ -0,0 +1,537 @@
+		return
+	if(rotate())
+		return TRUE
+	to_chat(user,"<span class='info'>Target area blocked, please clear all objects and personnel.<span/>")
```
```suggestion
	to_chat(user, "<span class='info'>Target area blocked, please clear all objects and personnel.</span>")
```

Migratingcocofruit replied:
done

Burzah commented on code/goonstation/modules/power/power_transmission_laser/transmission_laser.dm:
```diff
@@ -0,0 +1,537 @@
+
+/obj/machinery/power/transmission_laser/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>Laser currently has [unsent_earnings] unsent credits.<span/>"
+	. += "<span class='notice'>Laser has generated [total_earnings] credits.<span/>"
+	. += "<span class='notice'>Laser has sold [total_energy] Joules.<span/>"
```
```suggestion
	. += "<span class='notice'>Laser currently has [unsent_earnings] unsent credits.</span>"
	. += "<span class='notice'>Laser has generated [total_earnings] credits.</span>"
	. += "<span class='notice'>Laser has sold [total_energy] Joules.</span>"
```

Migratingcocofruit replied:
done

Migratingcocofruit commented on code/goonstation/modules/power/power_transmission_laser/transmission_laser.dm:
```diff
@@ -0,0 +1,537 @@
+
+/obj/machinery/power/transmission_laser/screwdriver_act(mob/living/user, obj/item/I)
+	if(firing)
+		to_chat(user,"<span class='info'>Turn the laser off first.<span/>")
```
done

Migratingcocofruit commented on code/goonstation/modules/power/power_transmission_laser/transmission_laser.dm:
```diff
@@ -0,0 +1,537 @@
+	var/list/targetable_areas = list(/area/lavaland/surface/outdoors/outpost,
+									/area/lavaland/surface/outdoors/targetable,
+									/area/mine/outpost,
+									/area/shuttle/mining)
```
done

Migratingcocofruit commented on code/goonstation/modules/power/power_transmission_laser/transmission_laser.dm:
```diff
@@ -0,0 +1,537 @@
+	origin_tech = "engineering=2;combat=3;"
+	req_components = list(
+							/obj/item/stock_parts/capacitor = 3,
+							/obj/item/stock_parts/micro_laser = 3)
```
done
